tests/unit/parsers/test.manifestjson.js

```diff
@@ -102,6 +102,125 @@ describe('ManifestJSONParser', () => {
     expect(warnings[0].code).toEqual('IGNORED_APPLICATIONS_PROPERTY');
   });
 
+  it.each([
+    {
+      title: 'hidden and browser_action',
+      manifest: {
+        hidden: true,
+        browser_action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden and action',
+      manifest: {
+        hidden: true,
+        action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden and page_action',
+      manifest: {
+        hidden: true,
+        page_action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden, browser_action and page_action',
+      manifest: {
+        hidden: true,
+        page_action: {},
+        browser_action: {},
+      },
+      privileged: true,
+      expectError: true,
+    },
+    {
+      title: 'hidden and browser_action but not privileged',
+      manifest: {
+        hidden: true,
+        browser_action: {},
+      },
+      privileged: false,
+      expectError: false,
+    },
+    {
+      title: 'hidden and page_action but not privileged',
+      manifest: {
+        hidden: true,
+        page_action: {},
+      },
+      privileged: false,
+      expectError: false,
+    },
+    {
+      title: 'hidden but no action',
+      manifest: {
+        hidden: true,
+      },
+      privileged: true,
+      expectError: false,
+    },
+    {
+      title: 'browser action but no hidden prop',
+      manifest: {
+        browser_action: {},
+      },
+      privileged: true,
+      expectError: false,
+    },
+    {
+      title: 'browser action but not hidden',
+      manifest: {
+        hidden: false,
+        browser_action: {},
+      },
+      privileged: true,
+      expectError: false,
+    },
+    {
+      title: 'page action but not hidden',
+      manifest: {
+        hidden: false,
+        page_action: {},
+      },
+      privileged: true,
+      expectError: false,
+    },
+  ])(
+    'should report an error when hidden and actions are defined - $title',
+    ({ manifest, privileged, expectError }) => {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON(manifest);
```

Comment:
I haven't checked if this is affecting somehow the test coverage as a not fully visible side-effect, it is likely not affecting it because the checks for these restrictions are completely independent from the schema validations, but to prevent our future selves to ask themselves the same question, I'm wondering if we should add into the manifest properties for the entries using `action` instead of `browser_action` an explicit `manifest_version: 3`, wdyt?